Core/src/org/sleuthkit/autopsy/datamodel/FileTypesByExtension.java

```diff
@@ -330,24 +331,39 @@ long calculateChildCount() throws TskCoreException {
         }
     }
 
-    private static String createQuery(FileTypesByExtension.SearchFilterInterface filter) {
-        StringBuilder query = new StringBuilder();
-        query.append("(dir_type = ").append(TskData.TSK_FS_NAME_TYPE_ENUM.REG.getValue()).append(")"); //NON-NLS
-        if (UserPreferences.hideKnownFilesInViewsTree()) {
-            query.append(" AND (known IS NULL OR known != ").append(TskData.FileKnown.KNOWN.getFileKnownValue()).append(")"); //NON-NLS
-        }
-        query.append(" AND (NULL"); //NON-NLS
-        for (String s : filter.getFilter()) {
-            query.append(" OR LOWER(name) LIKE LOWER('%").append(s).append("')"); //NON-NLS
+    private String createQuery(FileTypesByExtension.SearchFilterInterface filter) {
+        if (filter.getFilter().isEmpty()) {
+            return "";
```

Comment:
I'm leaning towards throwing an IllegalArgumentException if the filter list is empty since this is clearly a programming error that we want to be aware of. Thoughts?